### PR TITLE
Navbar Refactoring

### PR DESCRIPTION
**Pull request by wmarques**
Also removed the Language Controller as suggested in #3807.

The dropdowns are not working for the moment as we didn't implemented the ng2 ui bootstrap, so all of the ng1 directives won't work for now (ui-sref, active-menu, uib-dropdown...).

### CODE REVIEW

File: generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
                 <span class="sr-only">Toggle navigation</span>
                 <span class="icon-bar"></span>
                 <span class="icon-bar"></span>
                 <span class="icon-bar"></span>
             </button>
-            <a class="navbar-brand logo" href="#/" ng-click="vm.collapseNavbar()">
+            <a class="navbar-brand logo" href="#/" (click)="collapseNavbar()">
                 <img class="logo-img" src="content/images/logo-jhipster.png"/>
-                <span translate="global.title"><%= capitalizedBaseName %></span> <span class="navbar-version">v{{VERSION}}</span>
+                <span translate="global.title">Ng2jhi</span> <span class="navbar-version">v{{VERSION}}</span>
```
Comment (sendilkumarn) [inline generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html]: `Ng2jhi` ?

File: generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html [hunk 2]
```diff
@@ -95,91 +93,71 @@
...
                         <li ui-sref-active="active">
-                            <a ui-sref="logs" ng-click="vm.collapseNavbar()">
+                            <a ui-sref="logs" (click)="collapseNavbar()">
                                 <span class="glyphicon glyphicon-tasks"></span>&nbsp;
                                 <span translate="global.menu.admin.logs">Logs</span>
                             </a>
                         </li>
-                        <li ng-show="vm.swaggerEnabled" ui-sref-active="active">
-                            <a ui-sref="docs" ng-click="vm.collapseNavbar()">
+                        <li [hidden]="!swaggerEnabled" ui-sref-active="active">
```
Comment (sendilkumarn) [inline generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html]: ngIf please

File: generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html [hunk 3]
```diff
@@ -95,91 +93,71 @@
...
-                        <li ui-sref-active="active">
-                            <a ui-sref="<%=jhiPrefix%>-tracker" ng-click="vm.collapseNavbar()">
-                                <span class="glyphicon glyphicon-eye-open"></span>&nbsp;
-                                <span translate="global.menu.admin.tracker">User tracker</span>
-                            </a>
-                        </li>
-                        <%_ } _%>
                         <li ui-sref-active="active">
-                            <a ui-sref="<%=jhiPrefix%>-metrics" ng-click="vm.collapseNavbar()">
+                            <a ui-sref="jhi-metrics" (click)="collapseNavbar()">
```
Comment (sendilkumarn) [inline generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html]: can we follow the same here ? `<%=jhiPrefix%>`

**Comment by sendilkumarn** [comment]
good one 👍

File: generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html (hunk 2 above)
Comment (wmarques) [inline generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html]: `ngIf` and `[hidden]` does not do the same thing:
`ngIf` remove the element from the DOM while hidden just add a CSS property to hide the element, that's what `ng-show` does, that's why I put `hidden` and not `ngIf` there, is that ok for you ? :)

**Comment by deepu105** [comment]
I think I merged your other PR and that is causing conflict here, can you rebase plz?

**Comment by wmarques** [comment]
Dunno why tests are failing :(

**Comment by wmarques** [comment]
I don't know why I try to generate an app, Yeoman doesn't copy any files and just stops after prompting questions..

**Comment by deepu105** [comment]
There might be a silent error or an ejs error on one of the files being
copied

Thanks & regards,
Deepu

**Comment by wmarques** [comment]
Trying to check diff to see if there is an error like this

**Comment by wmarques** [comment]
Ok I had one ejs error in common module file... Quite to difficult to find it as Yeoman doesn't give any clue about the crash :(

Let's wait for tests but it should be fine, I hope !

**Pull request merged**